lib/src/rules/prefer_const_declarations.dart

```diff
@@ -80,6 +81,10 @@ class _Visitor extends SimpleAstVisitor<void> {
     if (!node.isFinal) return;
     if (node.variables.every((declaration) =>
         declaration.initializer != null &&
+        (declaration.initializer is! TypedLiteral ||
+            (declaration.initializer is TypedLiteral &&
+                declaration.initializer.beginToken?.keyword ==
+                    Keyword.CONST)) &&
```

Comment:
I'm sure there's a better way to do this?

Reply:
I'm not thinking of one at the moment, but the condition on line 85 appears to be redundant. Also, you might want to assign `declaration.initializer` to a local variable in order to shorten the code a bit.